requre/helpers/files.py

```diff
@@ -115,45 +152,45 @@ def _copy_logic(cls, cassette: Cassette, pathname: str, keys: list) -> None:
                         file_name=os.path.basename(pathname),
                         content=fileobj.getvalue(),
                     )
+                    serialized = serialization.to_serializable(return_value)
+                    output = {
+                        FILENAME: file_name,
+                        RETURNED: serialized,
+                        TARGET_PATH: pathname,
+                    }
                     cassette.store(
                         keys=cls.basic_ps_keys + keys,
-                        values=file_name,
+                        values=output,
                         metadata=metadata,
                     )
             finally:
                 os.chdir(original_cwd)
         else:
-            value = cassette[cls.basic_ps_keys + keys]
-            content = cls.read_file_content(cassette=cassette, file_name=value)
-            with BytesIO(content) as fileobj:
-                with tarfile.open(
-                    mode=f"r:{cls.tar_compression}", fileobj=fileobj
-                ) as tar_store:
-                    tarinfo_1st_member = tar_store.getmembers()[0]
-                    if tarinfo_1st_member.isfile():
-                        with open(pathname, mode="wb") as output_file:
-                            output_file.write(
-                                tar_store.extractfile(tarinfo_1st_member).read()
-                            )
-                    else:
-                        for tar_item in tar_store.getmembers():
-                            # we have to modify path of files to remove topdir
-                            if len(tar_item.name.split(os.path.sep, 1)) > 1:
-                                tar_item.name = tar_item.name.split(os.path.sep, 1)[1]
-                            else:
-                                tar_item.name = "."
-                            try:
-                                tar_store.extract(tar_item, path=pathname)
-                            except IOError:
-                                # rewrite readonly files if necessary
-                                os.remove(os.path.join(pathname, tar_item.name))
-                                tar_store.extract(tar_item, path=pathname)
+            output = cassette[cls.basic_ps_keys + keys]
+            pathname = pathname or output[RETURNED]
+            content = cls.read_file_content(
+                cassette=cassette, file_name=output[FILENAME]
+            )
+            # WORKAROUND: some tools uses old dir and some new one.
+            for item in [pathname, output[TARGET_PATH]]:
+                cls.__read_file(content, item)
```

Comment:
What `tools` do you mean? We haven't used this implementation anywhere so we can safely remove compatibility tweaks, can't we?

Reply:
@lachmanfrantisek e.g. in tests and yep, could be possible that we can remove it and tweak tests, but I've just added some comment to know why I'm doing this strange thing.
and maybe some packit code can use this way. probabaly it is caused by changes in this static tmp handling.
old way with counter avoid to touch this part. but this new way, introduce there this problem. and this fix this issue.